### PR TITLE
Docs: update migration guides

### PR DESCRIPTION
**Pull request by nataliaElv**
# Description

**Type of change**

- Documentation update

**Checklist**

- I added relevant documentation
- I followed the style guidelines of this project
- I did a self-review of my code
- I made corresponding changes to the documentation
- I confirm My changes generate no new warnings
- I have added tests that prove my fix is effective or that my feature works
- I have added relevant notes to the CHANGELOG.md file (See https://keepachangelog.com/)

### CODE REVIEW

**Comment by frascuchon** [comment]
We should add a settings tab depending [here](https://github.com/argilla-io/argilla/pull/5311/files#diff-d65989ca8a6908333ce0090536321b1eb45e039f97d33a14b257dcc89d4f7d9cL77) on the legacy dataset to migrate as we do for the records mapping. This allows us to define the expected question name when publishing records with responses and suggestions. What do you think @nataliaElv?

File: argilla/docs/how_to_guides/migrate_from_legacy_datasets.md [hunk 1]
```diff
@@ -68,30 +68,101 @@ client = rg.Argilla()
...
-```
+=== "For single-label classification"
+
+    ```python
+    settings = rg.Settings(
+        fields=[
+            rg.TextField(name="text"), # (1)
+        ],
+        questions=[
+            rg.LabelQuestion(name="label", labels=settings_v1.label_schema),
```
Comment (frascuchon) [inline argilla/docs/how_to_guides/migrate_from_legacy_datasets.md:79]: Should we add a description about the name here pointing to the record helper functions?

**Review by sdiazlor — COMMENTED** [review]
LGTM, just found this missing comment here: 
```python
# For
users_by_name = {user.username: user for user in client.users}
current_user = client.me
```
As a suggestion, to make the notes more direct, we could avoid the "Here, we need to" and go for the "We provide ..." as they are already placed next to the corresponding code snippet.

File: argilla/docs/how_to_guides/migrate_from_legacy_datasets.md (hunk 1 above)
Comment (nataliaElv) [reply argilla/docs/how_to_guides/migrate_from_legacy_datasets.md:79]: @frascuchon I've added tootips to the helper functions below. Check them out and let me know if you think that's enough.

File: argilla/docs/how_to_guides/migrate_from_legacy_datasets.md [hunk 2]
```diff
@@ -157,25 +244,41 @@ Here are a set of example functions to convert the records for single-label and
...
+                    agent=agent
+                )
+            )
 
         if annotation := data.get("annotation"):
             user_id = users_by_name.get(data["annotation_agent"], current_user).id
-            responses.append(rg.Response(question_name="label", value=annotation, user_id=user_id))
+            responses.append(
+                rg.Response(
+                    question_name="label", # (2)
```
Comment (frascuchon) [inline argilla/docs/how_to_guides/migrate_from_legacy_datasets.md]:
```suggestion
                    question_name="labels", # (2)
```

**Review by frascuchon — APPROVED** [review]

**Pull request merged**